gemini/use-cases/education/resume_generator.ipynb

```diff
@@ -0,0 +1,877 @@
+        "\n",
+        "Learn more about [setting up a project and a development environment](https://cloud.google.com/vertex-ai/docs/start/cloud-environment)."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "Nqwi-5ufWp_B"
+      },
+      "outputs": [],
+      "source": [
+        "# Use the environment variable if the user doesn't provide Project ID.\n",
+        "import os\n",
+        "\n",
+        "import vertexai\n",
+        "\n",
+        "PROJECT_ID = (\n",
+        "    \"\"  # @param {type: \"string\", placeholder: \"[your-project-id]\", isTemplate: true}\n",
+        ")\n",
+        "if not PROJECT_ID or PROJECT_ID == \"\":\n",
+        "    PROJECT_ID = str(os.environ.get(\"GOOGLE_CLOUD_PROJECT\"))\n",
+        "\n",
+        "LOCATION = os.environ.get(\"GOOGLE_CLOUD_REGION\", \"us-central1\")\n",
+        "\n",
+        "vertexai.init(project=PROJECT_ID, location=LOCATION)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "5303c05f7aa6"
+      },
+      "source": [
+        "### Import libraries"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "6fc324893334"
+      },
+      "outputs": [],
+      "source": [
+        "from ast import literal_eval\n",
+        "\n",
+        "# Native Imports\n",
+        "import os\n",
+        "\n",
+        "# Third-party Imports\n",
+        "import docx\n",
+        "import docx2txt\n",
+        "from docx.document import Document\n",
+        "from docx.oxml.table import CT_Tbl\n",
+        "from docx.oxml.text.paragraph import CT_P\n",
+        "from docx.table import Table, _Cell\n",
+        "from docx.text.paragraph import Paragraph\n",
+        "import pypdf\n",
+        "from vertexai.generative_models import GenerativeModel"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "5303c05f7aa6"
+      },
+      "source": [
+        "### Import a sample resume file"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "5c4978c07d9a"
+      },
+      "outputs": [],
+      "source": [
+        "! gsutil cp gs://github-repo/generative-ai/gemini/use-cases/education/resume-generation/Sample_1.docx ."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "e43229f3ad4f"
+      },
+      "source": [
+        "### Load model"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "cf93d5f0ce00"
+      },
+      "outputs": [],
+      "source": [
+        "MODEL_NAME = \"gemini-1.5-flash\"  # Choice of model\n",
+        "generation_config = {\"temperature\": 0}\n",
```

Comment:
Use the `GenerationConfig` class

Reply:
Replaced with GenerationConfig class and added controlled generation across the code